### ISSUE TITLE
'ovs remove nodes' leaves entry in AlbaNodesList

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework-alba-plugin#180]
# Problem description
Claiming disks from a reinstalled node, that has been previously removed, fails without any feedback to the user. 
There is no entry in the /var/log/upstart/ovs-workers.log about a 'claim disk task' nor a message on the gui reporting the problem to the user.
# Possible root of the problem
There is a lingering record of the previous installation in the AlbaNodesList because the ASD backend has not been uninstalled by 'ovs remove nodes' (Uninstallation not currently supported). All the calls might be forwarded to the Node with the old GUID instead of the new one.
# Possible solution
Adding a uninstall feature to 'ovs remove nodes'

----
# More details
###  Lingering records of the deleted node
Removing a node with a specified IP doesn't fully wipe all the data of the node. The information of that node persists in the GUI and in the AlbaNodesList
Visible on the GUI:
![nodepersistingui](https://cloud.githubusercontent.com/assets/17570109/17481985/ffe33824-5d7f-11e6-97d8-973099ff1f29.jpg)

Calling the API:
```
After removing node with id 'VVjJkReapIEM20VtfZNA3tck16ilw8UN' the node is still visible in the GUI and after calling the AlbaNodelist API we can see that its still in the list


In [3]: for i in AlbaNodeList.get_albanodes():
   ...:     print i
   ...:     
{
    "username": "root", 
    "password": "4YaXnpuKzT94cFVTQmQo8Azomyb96y1x", 
    "ip": "10.100.199.151", 
    "storagerouter_guid": "acad08ff-72e5-4d5e-8e63-627a14b3a480", 
    "ips": [], 
    "node_id": "VVjJkReapIEM20VtfZNA3tck16ilw8UN", 
    "guid": "5108952a-a186-4c36-9201-df6df7956e53", 
    "type": "ASD", 
    "port": 8500
}
{
    "username": "root", 
    "password": "q2sFx9kDy3BayGh8HeyhSwc9BEXwRsX5", 
    "ip": "10.100.199.152", 
    "storagerouter_guid": "40418014-1140-41bc-9208-69bbe3635d81", 
    "ips": [], 
    "node_id": "swDfLO2u0bDJnMCvjw0eVWpyaajWri74", 
    "guid": "63b08c01-4290-4b0b-aeda-d8431a452394", 
    "type": "ASD", 
    "port": 8500
}
{
    "username": "root", 
    "password": "zqoqyMTK5dzDJupcM5R637mk5UbND7oI", 
    "ip": "10.100.199.153", 
    "storagerouter_guid": "0c9785be-8840-4fe4-8744-6acf7fe7c65d", 
    "ips": [], 
    "node_id": "r8z3opfSPtatMSXpzM0q28RLsLN1ud6b", 
    "guid": "a7b0e146-6a13-465f-b507-ff39019bb014", 
    "type": "ASD", 
    "port": 8500
}
{
    "username": "root", 
    "password": "peunrkEOZTFqyKrfz6EBtPgpEC9OiY3q", 
    "ip": "10.100.199.151", 
    "storagerouter_guid": "acad08ff-72e5-4d5e-8e63-627a14b3a480", 
    "ips": [], 
    "node_id": "7dyeRphJPerBdiKEugENQINjPeizt1Ln", 
    "guid": "bc989538-b4eb-4216-80bf-6f2f0676bf14", 
    "type": "ASD", 
    "port": 8500
}
````

**Comment by khenderick** [comment]
Some technical details:
* ```ovs remove nodes ...``` removes the node from the framework, it also make sure alba asds/nodes are correctly removed from the backend if applicable.
* The asd-manager however, registers itself in etcd upon installation (at setup time), and the asd-manager has no uninstall support (neither has openvstorage - but that one is more or less wiped correctly by the (in)famous wiper), so this means the asd-manager's installation is still present in etcd. Even though the files/code themself might be wiped correctly, the etcd entry might still be there
* When the node is reinstalled, one of the tasks is searching for an asd-manager installation on the local node. This most likely detected both the old etcd entry and the entry of the new asd-manager installation, resulting in two asd-managers. This can cause various issues, of which the above might be one.

So I think this ticket can be considered a "support asd-manager uninstallation" request. In the mean time, the wiper should check for this etcd entry and remove it as well.

**Comment by JeffreyDevloo** [comment]
The following actions can be executed due to these lingering records:

- Initializing disks for the back-end
- Claim disks for the back-end

**Comment by khenderick** [comment]
As discussed, also as expected; the asd-manager is just doing whatever it's supposed to do; it has no knowledge about openvstorage, so it doesn't know something was removed.

**Comment by khenderick** [comment]
After some discussions with @JeffreyDevloo, it might be possible that a few side-effects from other bugs were messing with this ticket. @JeffreyDevloo, can you re-test this scenario so we know the behavior without the previously fixed issues?

**Comment by JeffreyDevloo** [comment]
Would be fixed when asd-manager would be removed. (The asd-manager should remove the entry from etcd)
Closing.

**Comment by JeffreyDevloo** [comment]
It must have been a side affect. I couldn't reproduce the issue.
Issue has been fixed for 2.7.2-rev.3859.42b3488-1
Closing.
